### DISCUSSION TITLE
Best practice for creating multiple component libraries using Tailwind

### DISCUSSION THREAD

**Discussion by nareshbhatia** [tailwindlabs/tailwindcss#11065] [Help]
I'd like to create multiple component libraries that build upon each other (atoms, molecules, organisms, ...). These libraries will serve as the foundation for multiple apps at our organization. I can think of two approaches to do this:

1. Run Tailwind CLI on an app and all its dependent libraries in one shot.
2. Give a unique prefix to each library and run the CLI on each one separately. This will result in multiple namespaced libraries. Some styles may overlap (e.g. layout styles), but the advantage is that all libraries are decoupled and the build process will be much simpler.

Which approach would you recommend and why? Any gotchas?

**Comment by nareshbhatia** [comment]
@adamwathan, any guidance please? What have you seen working well in these use cases? TIA

**Reply by AdsonCicilioti** [reply]
Hey man!

check the #11172
